packages/scoped-custom-element-registry/src/scoped-custom-element-registry.ts

```diff
@@ -120,762 +120,795 @@ type ParametersOf<
   T extends ((...args: any) => any) | undefined
 > = T extends Function ? Parameters<T> : never;
 
-const NativeHTMLElement = window.HTMLElement;
-const nativeDefine = window.customElements.define;
-const nativeGet = window.customElements.get;
-const nativeRegistry = window.customElements;
-
-const definitionForElement = new WeakMap<
-  HTMLElement,
-  CustomElementDefinition
->();
-const pendingRegistryForElement = new WeakMap<
-  HTMLElement,
-  ShimmedCustomElementsRegistry
->();
-const globalDefinitionForConstructor = new WeakMap<
-  CustomElementConstructor,
-  CustomElementDefinition
->();
-// TBD: This part of the spec proposal is unclear:
-// > Another option for looking up registries is to store an element's
-// > originating registry with the element. The Chrome DOM team was concerned
-// > about the small additional memory overhead on all elements. Looking up the
-// > root avoids this.
-const scopeForElement = new WeakMap<Node, Element | ShadowRoot>();
-
-class AsyncInfo<T> {
-  readonly promise: Promise<T>;
-  readonly resolve: (val: T) => void;
-  constructor() {
-    let resolve: (val: T) => void;
-    this.promise = new Promise<T>((r) => {
-      resolve = r;
-    });
-    this.resolve = resolve!;
-  }
-}
-
-// Constructable CE registry class, which uses the native CE registry to
-// register stand-in elements that can delegate out to CE classes registered
-// in scoped registries
-class ShimmedCustomElementsRegistry implements CustomElementRegistry {
-  private readonly _definitionsByTag = new Map<
-    string,
+if (typeof window === 'undefined') {
+  console.warn(
+    'The Scoped Custom Element Registry polyfill was imported in a non-browser environment. The polyfill will not be applied.'
+  );
```

Comment:
Here's the other major change.  In server environments, just log a warning but do nothing.  Otherwise, proceed with the rest.